### ISSUE TITLE
Not convert absolute path to relative in compiled codes

### ISSUE THREAD

**Issue by amirqasemi74** [nestjs/nest-cli#1770]
### Did you read the migration guide?

- [ ] I have read the whole migration guide

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Versions

9.0.0 -> 9.1.1.

### Describe the regression

I have set tsconfig.baseUrl in tsconfig.json to `src` to have imports with absolute path like `import { AuthModule } from 'core/auth/auth.module';` and it was working correctly in version 9.0.0. I had `const auth_module_1 = require("./core/auth/auth.module");` in compiled code in `dist` folder.

but we i upgraded to verison 9.1.1 it's compiling to `const auth_module_1 = require("core/auth/auth.module");` with no change in tsconfig and app does not run

### Minimum reproduction code

```ts

```

### Expected behavior

I expected to compiled with relative path like in 9.0.0: `const auth_module_1 = require("./core/auth/auth.module");`

**Comment by kamilmysliwiec** [comment]
Please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project).

**Comment by frbvianna** [comment]
Had this same issue as well for the v9.1.1 update.

We rely on this tsconfig.json configuration to allow non-relative import statements such as

`import config from 'src/configs/env.config'`: 
```json
{
  "compilerOptions": {
    "baseUrl": "./"
  }
}
```

Tested that this is seemingly happening due to this new section of code, introduced in https://github.com/nestjs/nest-cli/commit/01c5acd1:
https://github.com/nestjs/nest-cli/blob/01c5acd18351a8515c709a19a47425c0dd55ebc6/lib/compiler/hooks/tsconfig-paths.hook.ts#L14-L16
When I commented out these lines in the compiled code for nest-cli, I could compile my project again.

Also resolved if not commenting out those lines but adding `paths` to the TS compiler options as follows:
```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "src/*": ["src/*"]
    }
  }
}
````

Does this mean it is required to define `paths` in TS compiler options now? From our perspective, this would be an undocumented breaking change since it worked as intended before without `paths`, breaking the compilation entirely after the change.
